### ISSUE TITLE
Seperate button declaration and button style options

### ISSUE THREAD

**Issue by guyca** [wix/react-native-navigation#3792]
Currently buttons and their corresponding style options are declared together. Therefore, when setting buttons, The user also needs to be aware of the button style options.
We should separate button style options from the actual button declaration as styles can be determined dynamically outside of the screen scope.

**Suggested API**
The following options are applied to all right and left buttons
```js
topBar: {
  buttons: {
    right: {
      color: 'red',
      fontFamily: 'helveticaneue-condensedbold'
    },
    left: {
    }
  }
}
```
**Button Declaration**
```js
static get buttons() {
  right: [
    {
      id: 'btn1',
      color: 'green' // overrides color defined in topBar.right.color
    }
  ],
  left: []
}
```

```js
Navigation.setButtons(componentId, {
  right: [],
  left: [],
  fab: {}
});
```

**Comment by DanielZlotin** [comment]
1. Adding another static entry point is a bad idea. 1 is 1 too many. It's a react anti-pattern and we use it because we HAVE to.
2. Using dynamic styles (via mergeOptions) always happens after at least 1 render, so it can cause flickers and it's usually too late.

A better solution will be to add support to pass the `passProps` to the `static options` (making it a function instead of an object):
instead of:
```js
static get options { ... }
```
you'll have
```js
static options(passProps) { ... }
```

This will allow separation between the screen itself and it's invokers in terms of styles.
basically you can send whatever you want under passProps and let the screen use what it wants from there.

**Comment by DanielZlotin** [comment]
fixed with https://github.com/wix/react-native-navigation/commit/cf7e817f9631cede9cd3cfb7dd445c76e6ad03ab
